lua/entities/gmod_wire_expression2/core/findquery.lua

```diff
@@ -0,0 +1,261 @@
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:notInCone(vector origin, vector axis, number rad, number length)
+	this.cost = this.cost + 4
+
+	local sine = math.cos(rad)
+	this.filters[#this.filters + 1] = function(e)
+		return not util.IsPointInCone(e:GetPos(), origin, axis, sine, length)
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:inBox(vector min, vector max)
+	this.cost = this.cost + 2
+
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetPos():WithinAABox(min, max)
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:notInBox(vector pos, vector dir, number length, number rad)
+	this.cost = this.cost + 2
+
+	this.filters[#this.filters + 1] = function(e)
+		return not e:GetPos():WithinAABox(min, max)
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:withClass(string class)
+	this.cost = this.cost + 0.125
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetClass() == class
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:withoutClass(string class)
+	this.cost = this.cost + 0.125
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetClass() ~= class
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:withModel(string model)
+	this.cost = this.cost + 0.125
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetModel() == model
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:withoutModel(string model)
+	this.cost = this.cost + 0.125
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetModel() ~= model
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:withOwner(entity owner)
+	this.cost = this.cost + 0.125
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetOwner() == owner
+	end
+
+	return this
+end
+
+[nodiscard]
+e2function findquery findquery:withoutOwner(entity owner)
+	this.cost = this.cost + 0.125
+	this.filters[#this.filters + 1] = function(e)
+		return e:GetOwner() ~= owner
+	end
+
+	return this
+end
+
+__e2setcost(10)
+
+[nodiscard]
+e2function array findquery:query()
+	local all = ents.GetAll()
+	local out, nout, cost = {}, 0, this.cost
+
+	for _, ent in ipairs(all) do
+		self.prf = self.prf + 1 + cost
+		if self.prf > e2_tickquota then error("perf", 0) end
+
+		for _, filter in ipairs(this.filters) do
+			if not filter(ent) then
+				goto skip
+			end
+		end
+
+		nout = nout + 1
+		out[nout] = ent
+
+		::skip::
+	end
+
+	return out
+end
+
+[nodiscard]
+e2function array findquery:query(array entities)
+	local out, nout, cost = {}, 0, this.cost
+
+	for _, ent in ipairs(entities) do
+		self.prf = self.prf + 1.5 + cost
+		if self.prf > e2_tickquota then error("perf", 0) end
+
+		if not IsValid(ent) then
+			goto skip
+		end
+
+		for _, filter in ipairs(this.filters) do
+			if not filter(ent) then
+				goto skip
+			end
+		end
+
+		nout = nout + 1
+		out[nout] = ent
+
+		::skip::
```

Comment:
Fixes the `goto` linter warning.
```suggestion
		if not IsValid(ent) then continue end

		local shouldInclude = true

		for _, filter in ipairs(this.filters) do
			if not filter(ent) then
				shouldInclude = false
				break
			end
		end

		if shouldInclude then
			nout = nout + 1
			out[nout] = ent
		end
```